include/zephyr/senss/senss_datatypes.h

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSS_DATATYPES_H_
+#define SENSS_DATATYPES_H_
+
+#include <stdint.h>
+
+/**
+ * @brief Data Types
+ * @addtogroup senss_datatypes
+ * @{
+ */
+
+/**
+ * @struct senss_sensor_value_header
+ * @brief sensor value header
+ *
+ * Each sensor value data structure should have this header
+ *
+ * Here use 'base_timestamp' (uint64_t) and 'timestampe_delta' (uint32_t) to
+ * save memory usage in batching mode.
+ *
+ * The 'base_timestamp' is for readings[0], the 'timestamp_delta' is relation
+ * to the previous 'readings'. So,
```

Comment:
What is the benefit in providing a per sample time delta vs a single delta or sample rate in the header here?

Reply:
Per sample time stamp or delta time is must needed, we can't assume the sample rate is fixed. 

Because, except report interval, the `sensitivity` also can impact the real sample rate. 

For example, for a pedometer sensor, we can set it's report interval as 1 second, if it's `sensitivity` is set to 0, then pedometer sensor will report data for every 1 second (including unchanged data samples), if `sensitivity` is 10,  it means every 10 step count detected then report the sample, so, the samples sequences will be t1:10, t2:20, t3:30, t4:40, ... etc, the time delta between these samples will not 1 second, and even not determined, for example, after t2, user may stopped walking for several minutes.